### ISSUE TITLE
modifying model spec in-place is not writing changes

### ISSUE THREAD

**Issue by jfree-man** [canmod/macpan2#178]
`mp_tmb_insert`/`mp_tmb_update` doesn't write changes in-place. If you save modifications in a new model spec object, then changes are saved. (this is with the latest version of macpan2 installed, macpan2_1.2.0)

```r
si = mp_tmb_library("starter_models", "si", package = "macpan2")
print(si)
# this prints the changes we want automatically
mp_tmb_insert(si, phase = "before"
              , at = 1
              , expressions = list(beta ~ exp(log_beta))
              , default = list(log_beta = log(0.5))
)
# when we print again, the si object doesn't have the changes from mp_tmb_insert
print(si)
# we can save and this works
si_saved = mp_tmb_insert(si, phase = "before"
                         , at = 1
                         , expressions = list(beta ~ exp(log_beta))
                         , default = list(log_beta = log(0.5))
)
print(si_saved)
```

**Comment by stevencarlislewalker** [comment]
Thanks for catching this Jen.  I made this change but didn't tell you (or write it down?). Can you please add to the NEWS.md as a breaking change if not already there?  Can you also please add to the help file to clarify this?  Can you also adjust any existing scripts to make sure that they have a return value if needed?

The rationale for this change is that in-place changes can be confusing and I only want to do it when there is a benefit (e.g. for a calibrator object, when you want to try out different optimizers and optimizer settings without having to regenerate the calibrator object -- a potentially expensive operation).  Does this make sense?

**Comment by stevencarlislewalker** [comment]
Please close this issue as complete when the above doc changes are made.

**Comment by jfree-man** [comment]
Makes sense, will do.

**Comment by stevencarlislewalker** [comment]
Thank you

**Comment by jfree-man** [comment]
closed with commit 437b444
